src/Dom/Document.php

```diff
@@ -666,6 +668,45 @@ private function maybe_restore_noscript_elements( $html ) {
 		);
 	}
 
+	/**
+	 * Replaces the tag name of script[template="amp-mustache"], as a workaround to a parsing issue.
+	 *
+	 * This script can have closing tags of its children table and td stripped.
+	 * So this changes its name from script to tmp-script to avoid this.
+	 *
+	 * @link https://github.com/ampproject/amp-wp/issues/4254
+	 * @see restore_mustache_templates() Reciprocal function.
+	 *
+	 * @param string $html To replace the tag name of the mustache templates in.
+	 * @return string The HTML, with the tag name of the mustache templates replaced.
+	 */
+	private function replace_mustache_templates( $html ) {
+		return preg_replace(
+			'#<script(\s[^>]*template="amp-mustache"[^>]*)>(.*?)</script>#s',
```

Comment:
This can be made a bit more robust. Consider attribute values using single quotes or not quotes at all.

Reply:
Sure, I'll add handling for cases like `template='amp-mustache'` or `template=amp-mustache`

Reply:
What do you think about https://github.com/ampproject/amp-wp/pull/4333/commits/83d3dd6f2f829ee8dfbdadeba8d1842a672bc61b ?